nuxeo-features/nuxeo-platform-comment/nuxeo-platform-comment/src/main/java/org/nuxeo/ecm/platform/comment/impl/AbstractCommentManager.java

```diff
@@ -114,8 +113,11 @@ protected void notifyEvent(CoreSession session, String eventType, DocumentModel
                 }
             }
         }
-        DocumentEventContext ctx = new DocumentEventContext(session, principal, commentedDoc);
+        DocumentRef topLevelDocRef = getTopLevelCommentAncestor(session, commentedDoc.getRef());
+        DocumentModel topLevelDocument = session.getDocument(topLevelDocRef);
+        DocumentEventContext ctx = new DocumentEventContext(session, principal, topLevelDocument);
```

Comment:
Event should remain on `commentedDoc`.

Reply:
Then I need to alter the creation of the notification as it takes the source document as the commented document which we want to always be the top level document.
The source document is the one linked in the notification we are trying to fix.

Reply:
If it's the only solution, probably. But we need to keep current events on the comment for business liseners.

Reply:
Now catching the failing method and putting the ancestor from the DocumentEventContext. Allows to fix things and keep checking the mail count without changing the sourceDoc